source/mqtt_agent.c

```diff
@@ -344,81 +367,58 @@ static MQTTStatus_t createCommand( CommandType_t commandType,
-                      ( pMqttAgentContext != NULL ) &&
-                      ( isSpace == true );
-
-            break;
-
         case UNSUBSCRIBE:
+            assert( pMqttInfoParam != NULL );
 
             /* This message type results in the broker returning an ACK.  The
              * agent maintains an array of outstanding ACK messages.  See if
              * the array contains space for another outstanding ack. */
             isSpace = isSpaceInPendingAckList( pMqttAgentContext );
 
             pSubscribeArgs = ( MQTTAgentSubscribeArgs_t * ) pMqttInfoParam;
-            isValid = ( pMqttAgentContext != NULL ) &&
-                      ( pSubscribeArgs != NULL ) &&
-                      ( pSubscribeArgs->pSubscribeInfo != NULL ) &&
-                      ( isSpace == true );
+            isValid = isSpace;
+
             break;
 
         case PUBLISH:
             pPublishInfo = ( MQTTPublishInfo_t * ) pMqttInfoParam;
 
             /* Calculate the space consumed by everything other than the
              * payload. */
-            if( pPublishInfo == NULL )
+            uxHeaderBytes = uxControlAndLengthBytes;
+            uxHeaderBytes += pPublishInfo->topicNameLength;
+
+            /* This message type results in the broker returning an ACK. The
+             * agent maintains an array of outstanding ACK messages.  See if
+             * the array contains space for another outstanding ack.  QoS0
+             * publish does not result in an ack so it doesn't matter if
+             * there is no space in the ACK array. */
+            if( pPublishInfo->qos != MQTTQoS0 )
             {
-                isValid = false;
+                isSpace = isSpaceInPendingAckList( pMqttAgentContext );
             }
-            else
-            {
-                uxHeaderBytes = uxControlAndLengthBytes;
-                uxHeaderBytes += pPublishInfo->topicNameLength;
-
-                /* This message type results in the broker returning an ACK. The
-                 * agent maintains an array of outstanding ACK messages.  See if
-                 * the array contains space for another outstanding ack.  QoS0
-                 * publish does not result in an ack so it doesn't matter if
-                 * there is no space in the ACK array. */
-                if( pPublishInfo->qos != MQTTQoS0 )
-                {
-                    isSpace = isSpaceInPendingAckList( pMqttAgentContext );
-                }
 
-                isValid = ( pMqttAgentContext != NULL ) &&
-                          /* Will the message fit in the defined buffer? */
-                          ( ( pPublishInfo->payloadLength + uxHeaderBytes ) < pMqttAgentContext->mqttContext.networkBuffer.size ) &&
-                          ( isSpace == true );
-            }
+            /* Will the message fit in the defined buffer? */
+            isValid = ( ( pPublishInfo->payloadLength + uxHeaderBytes ) < pMqttAgentContext->mqttContext.networkBuffer.size ) &&
```

Comment:
I think the payload will not be copied over to the MQTT network buffer for publish. Do we need to consider the payload length in that case?

Reply:
Yes that's true, so I guess not